### PR TITLE
Undeclared properties should be allowed on Open Types

### PR DESCRIPTION
**Pull request by AdamCaviness**
Added support to ClientEdmModel to consult IEdmEntityType's IsOpen
property when instantiating EdmEntityTypeWithDelayLoadedProperties and
EdmComplexTypeWithDelayLoadedProperties.  ClientEdmModel was always
setting isOpen to false for these instances. This fixes an issue where
sending arbitrary/undeclared ODataProperty instances up would throw in
WriterValidationUtil.ValidatePropertyDefined().  This addresses
https://github.com/OData/odata.net/issues/143

### CODE REVIEW

**Comment by congysu** [comment]
@AdamCaviness Thanks very much for your contribution.  Not sure if you received email from our Pull Request Bot.  In case you didn't, please help signing a contribution license agreement: https://cla.microsoft.com.

**Comment by AdamCaviness** [comment]
@LaylaLiu, thanks for the great review.  I will work on these items.  Thank you!

**Comment by AdamCaviness** [comment]
@LaylaLiu I've made changes to address each issue you mentioned and again I thank you for your input.

File: test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientOpenTypeUpdateTests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ClientUpdateTests.cs" company="Microsoft">
```
Comment (LaylaLiu) [inline test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientOpenTypeUpdateTests.cs]: Change the file name

Comment (AdamCaviness) [inline test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientOpenTypeUpdateTests.cs]: @LaylaLiu Good catch, sorry about that.

**Comment by LaylaLiu** [comment]
Have merged by commit https://github.com/OData/odata.net/commit/39b01c17c871901eb0b3cedd21e035402cdc2358

**Pull request closed without merging**